Makefile

```diff
@@ -18,7 +18,7 @@ PATH:=$(PATH):$(PWD)/hack
 
 # docker image publishing options
 DOCKER_PUSH?=false
-IMAGE_TAG?=
+IMAGE_TAG?=$(shell git describe --dirty --always)
```

Comment:
I don't think this will play nicely with `hacks/generate-manifests.sh`, do you want to take a look?

Reply:
Do you mean `hack/update-manifests.sh` ? It seems depends on environment variable. If `IMAGE_TAG` is exported, it will take precedence in both Makefile and the script.

Anyway, I shall exclude unrelated changes for now and submit another patch to amend Contributing guide after a walk through. What do you think?